packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -331,6 +331,8 @@ export interface ITreeCheckoutFork extends ITreeCheckout {
  * An implementation of {@link ITreeCheckoutFork}.
  */
 export class TreeCheckout implements ITreeCheckoutFork {
+	private isDisposed = false;
```

Comment:
Do we need another piece of state for this? TreeCheckout already "fails" after disposal in most of the below methods because it tries to modify the branch. And the methods where it doesn't, like `locate` seem fine to call even if after disposal, since they are read only.

Put another way - TreeCheckout's current notion of "disposed" is completely derived from whether or not its branch is disposed. Seems kind of nice - I don't see a reason to change that (yet). 

I like replacing the `dispose` method with the symbol though.

Reply:
I'm not sure what's best here. I think it's plausible we may want to prevent things like reading from the forest. Right now we allow it but that's because we don't dispose the forest, which we may want to do in the future. Similarly, querying anchors seems questionable. So, I think you're right that, right now, relying on the branch would be sufficient, but I think it may be incidental. Also, as a user, I'd rather get an error on `checkout.editor` than deep down in my editing logic.

Reply:
I agree about the errors - but this class is still internal. I don't really care much one way or the other until we make it public, or make something else like it public. I'll defer to you since you're working in this space.